sklearn/impute.py

```diff
@@ -519,6 +539,9 @@ def _get_missing_features_info(self, X):
                 np.flatnonzero(np.diff(missing_values_mask.indptr))
                 if missing_values_mask.format == 'csc'
                 else np.unique(missing_values_mask.indices))
+            imputer_mask, features_with_missing = self._drop_constant_columns(
```

Comment:
This changes public API doesn't it? Doesn't this correspond to adding a different value for the `MissingIndicator.features` parameter?

Reply:
Yes, now columns with only missing values are dropped, as in the `SimpleImputer`. The current options for the `features` parameter are keep all columns or drop columns with no missing.

We could make it another option but we don't in the imputer and columns full of missing bring no information.

Reply:
Well, I'm not convinced we should be changing it without warning. It won't affect usage very often, though.

Reply:
yeah looks like it would require a deprecation cycle. If we add an option and pass the option in ``SimpleImputer`` then it's unlikely it'll affect a lot of users, I think.
Most people will probably use the convenience in ``SimpleImputer``.

Reply:
It's unlikely to affect a lot of users in any case... It's dealing with extremely degenerate data